src/core/tsi/transport_security.h

```diff
@@ -111,6 +111,9 @@ tsi_result tsi_construct_allocated_string_peer_property(
 tsi_result tsi_construct_string_peer_property_from_cstring(
     const char *name, const char *value, tsi_peer_property *property);
 
+/* Destroy plugin registration funcion. */
+void tsi_register_destroy_plugin(void (*destroy)(void));
```

Comment:
Regarding the first issue, if you think dependency is not a problem, I think I will use the existing grpc plugin mechanisms instead of creating a new registration mechanism. I will also remove tsi_init and tsi_destroy from TSI since corresponding TSI implementation APIs will be registered to grpc directly, instead of going through TSI.